src/libsystemd/sd-journal/journal-file.c

```diff
@@ -336,6 +336,37 @@ static bool compact_mode_requested(void) {
         return true;
 }
 
+static Compression compression_requested(void) {
+#if HAVE_COMPRESSION
+        const char *e;
+        Compression c;
+        int r;
+
+        e = getenv("SYSTEMD_JOURNAL_COMPRESS");
+        if (!e)
+                return DEFAULT_COMPRESSION;
+
+        r = parse_boolean(e);
```

Comment:
This makes certain sense, but is it really needed? We already have a config. file option (`Compress=`) for this, so it doesn't bring anything new. IMHO the new env. var. should be used just for selection of the compression alg. (maybe rename it to `SYSTEMD_JOURNAL_COMPRESSION_METHOD` or so?)

Reply:
This is mostly for `COMPRESSION_NONE`. We can set it with "NONE", but maybe arbitrary negative boolean value is helpful.

Also, we sometimes have redundant options: e.g. `bootctl --esp-path` and `$SYSTEMD_ESP_PATH`.

Reply:
> This is mostly for `COMPRESSION_NONE`. We can set it with "NONE", but maybe arbitrary negative boolean value is helpful.

But it also accidentally accepts `yes`/`no`. And how useful is allowing `none` anyway? It requires editing of a conf. file just like the existing way, and it's longer: `Environment=SYSTEMD_JOURNAL_COMPRESSION=none` vs. `Compress=no`.

> Also, we sometimes have redundant options: e.g. bootctl --esp-path and $SYSTEMD_ESP_PATH.

Yes, but these have a different scope. The command line option is for one-off use, while the env. var. can be set in environment.

Reply:
I mean, I don't really care that much. It just seems to me you're overthinking this a bit.

Reply:
Note, the scope _is_ different. `Compress=yes|no` works only for journald. But `SYSTEMD_JOURNAL_COMPRESS=yes|no|XZ|LZ4|ZSTD` also works journal-remote.
(Maybe, we should also introduce `Compress=yes|no` for journal-remote.conf later.)